pkg/cloud/aws/services/kubeadm/aws_defaults.go

```diff
@@ -162,6 +166,11 @@ func SetJoinNodeConfigurationOverrides(caCertHash, bootstrapToken string, machin
 		klog.Infof("Overriding node's cloud-provider to the required value of %q.", cloudProvider)
 	}
 	base.NodeRegistration.KubeletExtraArgs["cloud-provider"] = cloudProvider
+	if machine.Machine.Labels["set"] == "node" {
+		base.NodeRegistration.KubeletExtraArgs["node-labels"] = nodeRole
+	} else {
+		base.NodeRegistration.KubeletExtraArgs["node-labels"] = controlPlaneRole
```

Comment:
Is setting controlPlaneRole necessary here? `kubeadm` should handle this for us.